instana-java-sdk/src/main/java/com/instana/sdk/annotation/TagParam.java

```diff
@@ -0,0 +1,72 @@
+package com.instana.sdk.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Arrays;
+
+/**
+ * <p>
+ * This annotation enables you to declare that the value of a method parameter it marks must be stored as a tag on
+ * the currently active span. This can be an Instana AutoTrace span or a manually created SDK span.
+ * <p>
+ * For example, consider the following parameter annotation:
+ * </p>
+ * <pre><code>
+ * public void getVeryImportantData(@TagParam("my.awesome.tag") Object importantParameter) {
+ * ...
+ * }
+ * </code></pre>
+ * <p>
+ * The value of the new tag {@code my.awesome.tag} will be the string representation of the parameter passed to the
+ * invocation of the method annotated with this annotation. More details on how the tag value is build are stated below.
+ * You can then search this tag and its value in Instana using the {@code call.tag} filter in Unbounded Analytics.
+ * </p>
+ * <p>
+ * The value of the resulting tag is build according to the following rules:
+ * </p>
+ * <p>
+ * The value of the new tag will be the result of the {@code #toString()} method invoked on the annotated parameter.
+ * Primitive types will be converted to a String. Arrays will be formatted by invoking {@link Arrays#toString(Object[])}
+ * on them.
+ * </p>
+ * <p>
+ * If the value of the annotated parameter is {@code null}, the tag will not be created.
+ * </p>
+ * <p>
+ * If the type of the annotated parameter is {@link java.util.Optional}, and the method
+ * {@link java.util.Optional#isPresent()} invoked on the passed in value returns {@code true}, the value of the
+ * resulting tag will be the string representation of the result of {@link java.util.Optional#get()}, as described above.
+ * If {@link java.util.Optional#isPresent()} returns {@code false}, the tag will not be added to the currently active span.
```

Comment:
Mention that this could trigger side-effects in Optionals that use generators or suppliers.

Reply:
Is that possible with Java `Optional`s? Only a concrete value can be provided, so generators should be no problem.

Reply:
Discussed in Slack, ok as it is.